src/composables/useSearchParams.ts

```diff
@@ -16,25 +30,96 @@ const searchQuery = z.object({
   }),
 });
 
+export function supportRangeOption (field: string) {
+  const mapping: Record<string, string> = {
+    created_dt: 'dateRange',
+    updated_dt: 'dateRange'
+  };
+  return mapping[field];
+}
+
 export function useSearchParams() {
 
   const route = useRoute();
 
   const router = useRouter();
 
+  function getRangeFromURL(value: string, key: string) {
+    let type: DateRange | null = null;
+    let start = null;
+    let end = null;
+    if (supportRangeOption(key)) {
+      const values = value.split('_');
+      if (values.length > 0) {
+        type = values[0] as DateRange;
+        if (values[1] && values[2]) {
+          start = supportRangeOption(key) === 'dateRange' ? new Date(values[1]).toISOString() : values[1];
+          end = supportRangeOption(key) === 'dateRange' ? new Date(values[2]).toISOString() : values[2];
+          type = DateRange.CUSTOM;
+        }
+      }
+    }
+    return {
+      type,
+      start,
+      end
+    };
+  }
+
+  function parseValueForURL(facet: Facet) {
+    const { field, value, range } = facet;
+    if (supportRangeOption(field)) {
+      const values = [range.type];
+      if (range.type === DateRange.CUSTOM) {
+        if (range.start) {
+          values.push(
+            // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+            //@ts-ignore
+            supportRangeOption(field) === 'dateRange' ? parseDate(range.start) : range.start
+          );
```

Comment:
I think we can rewrite the `facet`  bit so we don't need this.
Maybe using `range?` on the facet type and then asserting that `range` exists on `parseValueForURL`